pir/cpp/server.h

```diff
@@ -0,0 +1,69 @@
+//
+// Copyright 2020 the authors listed in CONTRIBUTORS.md
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#ifndef PIR_SERVER_H_
+#define PIR_SERVER_H_
+
+#include <string>
+#include <vector>
+
+#include "context.h"
+#include "database.h"
+#include "seal/seal.h"
+#include "util/statusor.h"
+
+namespace pir {
+
+using ::private_join_and_compute::StatusOr;
+
+class PIRServer {
+ public:
+  /**
+   * Creates and returns a new server instance, holding a database.
+   * @param[in] db Database to load
+   * @returns InvalidArgument if the database encoding fails
+   **/
+  static StatusOr<std::unique_ptr<PIRServer>> Create(
+      const std::vector<std::int64_t>& /*database*/);
```

Comment:
Should this not take in a DB instance as well as a context?

Reply:
wouldn't we assume that const std::vector<std::int64_t>&  is the database parameter, and the db class is for the internal representation?

I can merge the database class with the server for now, it has little use the current separation